### PR TITLE
dhcrelay4: add support for up/downstream interfaces

### PR DESCRIPTION
**Pull request by brianjmurrell**
Maintainer: @pprindeville
Compile tested: wndr4300, 17.01.4
Run tested: wndr4300, 17.01.4

Description:
dhcrelay has the ability to monitor interfaces for requests
in a single direction only rather than listening to all
interfaces for requests.

Doing this allows one to suppress the duplication of having
the relay forward requests from the same network that the
DHCP server is on.

Signed-off-by: Brian J. Murrell <brian@interlinx.bc.ca>

### CODE REVIEW

**Comment by brianjmurrell** [comment]
Can anyone do a review of this?

**Comment by brianjmurrell** [comment]
A review, please?

**Comment by champtar** [comment]
Hi @brianjmurrell 
maintainer is @pprindeville 
https://github.com/openwrt/packages/blob/master/net/isc-dhcp/Makefile#L17
see my (partial) review

**Comment by brianjmurrell** [comment]
> you need to increase PKG_RELEASE

Ooops.  Yes, of course.  Done.  Also updated the maintainer on the original comment.

**Comment by pprindeville** [comment]
Please squash and rebase.

**Comment by brianjmurrell** [comment]
> Please squash and rebase.

I will, but just for my future reference, can I ask why?  Since GH says it has no conflicts with the base branch, it should land cleanly, so I just want to understand why so that I will know for the future.

**Comment by pprindeville** [comment]
So that the PR is a single commit.  There's not enough complexity to warrant two commits.

**Comment by brianjmurrell** [comment]
@pprindeville Finally found some time to squash and rebase.  If you (or anyone else who cares to) could re-review, it would be much appreciated.

**Comment by brianjmurrell** [comment]
Hello?  Review?  Please?

**Comment by karlp** [comment]
Looks generally sane to me, leave it to maintainer to decide.

**Comment by pprindeville** [comment]
> Hello? Review? Please?

Trying to update to 4.4.0 and as soon as I have that done, will merge.

**Comment by pprindeville** [comment]
Can you test https://github.com/openwrt/packages/pull/5579 and confirm that it's not broken?

**Comment by pprindeville** [comment]
No one has had a change to test PR #5579?

File: net/isc-dhcp/files/etc/config/dhcrelay [hunk 1]
```diff
@@ -7,6 +7,8 @@ config dhcrelay ipv4
 
 	# network interfaces to listen on (e.g. lan or wan)
 	option 'interfaces' ''
+	#option 'upstream_interfaces' ''
```
Comment (pprindeville) [inline net/isc-dhcp/files/etc/config/dhcrelay]: Why does this option need to be commented out for an empty list?

Wouldn't an empty list result in zero iterations through the `for` loop?

Comment (brianjmurrell) [reply net/isc-dhcp/files/etc/config/dhcrelay]: [ Oops.  Just noticed this needed my attention. ]

Indeed.  It could just be empty and uncommented out.

**Pull request merged**